### PR TITLE
Fix notifications subscription leak when in narrow mode

### LINKED ISSUE #16952: Memory leak when keeping the dashboard open on a tablet or mobile (narrow) device
### The problem

The memory leak seems to have started after updating HAOS to 10.2 or core 2023.6 beta (coming from a dev build in the early stage of 2023.5, so it could also be related to a late 2023.5 specific version).
What I have noticed is that my memory increase is caused by my tablet which shows the dashboard 24/7. 
I can reproduce the start of the memory leak and stop it aswell after some testing.
Bdraco had a look in my (profiler) logs and he said there are no obvious task leaks in the them, so I'm not sure if it is an issue in core or frontend. 

image
At the first drop in memory I did a restart of Home Assistant, but didn't refresh the webapp running on my tablet.
This causes the memory to increase as you can see. Around 08:00 I restarted the webapp on the tablet.
The memory stays the same, so no memory leak visible.
Around 09:00 I did another restart of Home Assistant without restarting the webapp on the tablet and there is the memory leak again.

This is also reported in the logs:
`[139928754006096] Tablet from  (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 Edg/114.0.1823.43): Client unable to keep up with pending messages. Reached 4096 pending messages. The system's load is too high or an integration is misbehaving; Last message was: {"id":22632,"type":"result","success":true,"result":null} [139928813456528] Tablet from  (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 Edg/114.0.1823.43): Client unable to keep up with pending messages. Reached 4096 pending messages. The system's load is too high or an integration is misbehaving; Last message was: {"id":20399,"type":"result","success":true,"result":null} [139928532840656] Tablet from  (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 Edg/114.0.1823.43): Client unable to keep up with pending messages. Reached 4096 pending messages. The system's load is too high or an integration is misbehaving; Last message was: {"id":20538,"type":"result","success":true,"result":null} [139928688859152] Tablet from  (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 Edg/114.0.1823.43): Client unable to keep up with pending messages. Reached 4096 pending messages. The system's load is too high or an integration is misbehaving; Last message was: {"id":70314,"type":"result","success":true,"result":null} [139928776142096] Tablet from  (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 Edg/114.0.1823.43): Client unable to keep up with pending messages. Reached 4096 pending messages. The system's load is too high or an integration is misbehaving; Last message was: {"id":30737,"type":"result","success":true,"result":null}`

### What version of Home Assistant Core has the issue?

core-2023.6.2

### What was the last working version of Home Assistant Core?

core-2023.5.0?

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

_No response_

### Link to integration documentation on our website

_No response_

### Diagnostics information

_No response_

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

In narrow mode (inspect, toggle into device mode) this is observed on the console
Screenshot 2023-06-17 at 6 05 36 PM

tested with mobile ✅  always hide sidebar ✅  normal ✅

fixes #16952

`oldNarrow` was always false every time `hass` changed so it would always make a new subscription.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bdraco** [comment]
I downgraded to 2023.5.4 and its calling `subscribeNotifications` over and over there as well.

**Comment by bdraco** [comment]
The impact of this was much lower in 2023.5.4 because `getCollection` will return the same collection if you ask to subscribe twice https://github.com/home-assistant/home-assistant-js-websocket/blob/a418262624c68347e654a2e3f3095db3b4e12c7d/lib/collection.ts#L32

**Comment by bdraco** [comment]
When #16476 migrated to using the `persistent_notification/subscribe` this problem has a much higher impact because it does not have the same logic to prevent multiple subscriptions

**Comment by bdraco** [comment]
I still haven't been able to figure out where the narrow check actually broke here though

File: src/components/ha-menu-button.ts [hunk 1]
```diff
@@ -98,6 +103,9 @@ class HaMenuButton extends LitElement {
   }
 
   private _subscribeNotifications() {
+    if (this._unsubNotifications) {
+      throw new Error("Already subscribed");
+    }
```
Comment (bdraco) [inline src/components/ha-menu-button.ts:108]: This might be overkill but it should ensure this never regresses

**Review by balloob — APPROVED** [review]

**Pull request merged**